### DISCUSSION TITLE
Feature request: transclusion

### DISCUSSION THREAD

**Discussion by ados8** [docmost/docmost#1623] [Ideas]
Recently discovered this platform and trying it out, the layout is excellent.
I'm currently using Bookstack and looking to switch but there's a few features I would miss.
Won't miss the search though, Docmost is way better. 👍

Currently Bookstack supports linking document content, this isn't URLs to pages.
This can be used with any content but for this I'll use a code example.
How does one maintain multiple pages using the same source?

You could copy the code across all the pages that reference it.
However, now you have multiple sources to track and update.

You could have a URL for each page linking back to a single page which has the content.
Then at least you have a single point when you want to update or make changes.
However this isn't as user friendly and less seamless.

A dynamic option is the ability to link paragraphs of the pages itself.
In Bookstack you highlight the portion of page (entire page possible) and copy the link.
At the reference page you paste the link and now the content is linked.
The reference page looks as if the text was pasted but when you update the source the reference content instantly updates as well.

Would this be something that might be implemented?

**Comment by Shadowfita** [comment]
I have been testing docmost over the last couple of days and had this same thought. Re-usable components / blocks would be extremely useful, especially if they were editable and that state updated at the source, similar to Microsoft Loop and others.

*EDIT*

I've found [this community extension for tiptap](https://github.com/sereneinserenade/tiptap-snippets-extension) which allows the creation of snippets based on the selected editor context. 

It could be used as a starting point for document linking, and a step further could be the creation of a custom tiptap node that retrieves the content from the server for displaying, manipulation, or easy deletion, from the source document.

https://github.com/sereneinserenade/tiptap-snippets-extension

**Comment by Shadowfita** [comment]
@Philipinho is this something you would accept as a PR?

**Comment by Philipinho** [comment]
@Shadowfita, this issue is about content "transclusion", I do not think it is worth the effort yet. I may consider it in the future.

Could you consider working on internal linking enhancements at https://github.com/docmost/docmost/issues/93?

**Comment by Philipinho** [comment]
Synced blocks "transclusion" now supported https://github.com/docmost/docmost/pull/2163.
